### PR TITLE
Make BbFlip on CPU act as on GPU

### PR DESCRIPTION
**Pull request by awolant**
CPU version of `BbFlip` was filtering boxes outside of the image. That's why it did not work with `SSDRandomCrop`.  Now both versions act the same in this manner. 

Signed-off-by: Albert Wolant <awolant@nvidia.com>

### CODE REVIEW

**Comment by JanuszL** [comment]
Maybe dali/test/python/test_random_bbcrop.py could be extended (and possibly renamed) to cover this change as well.

**Comment by awolant** [comment]
This change doesn't touch random cropping and there is no flip in dali/test/python/test_random_bbcrop.py 
I was planning to look exactly why ssd random crop gives boxes outside of images later and then maybe adjust this test, if necessary?
I can also maybe add python test for flipping?

**Comment by awolant** [comment]
Build 675871

**Comment by JanuszL** [comment]
> This change doesn't touch random cropping and there is no flip in dali/test/python/test_random_bbcrop.py
> I was planning to look exactly why ssd random crop gives boxes outside of images later and then maybe adjust this test, if necessary?
> I can also maybe add python test for flipping?

Sure. I just just thinking about testing CPU and GPU version side by side. Not necessarily using the same script as for bbox crop.

**Comment by awolant** [comment]
Right. I will add this kind of test.

**Comment by szalpal** [comment]
I'm wondering, in what circumstances we would like the BbFlip to work on bbs that are (even partially) outside of the image? 
Maybe we're should consider adding DALI_ENFORCE to runimpl guarding from this condition? 

I certainly wouldn't expect invalid bb on the input...

**Comment by awolant** [comment]
I added different flips to `dali/test/python/test_random_bbcrop.py` and renamed it as `dali/test/python/test_detection_pipeline.py`
I will refactor this test in the separate PR - I will add other operators from detection pipeline (ex. box encoder) and add testing against reference implementations from PyTorch. This will give us pretty comprehensive test for SSD related operators.

I also fixed BbFlip on GPU for empty boxes - it used to throw `Floating point exception` due to division by 0.

File: dali/test/python/test_detection_pipeline.py [hunk 1]
```diff
@@ -56,7 +72,7 @@ def define_graph(self):
         images, bb, labels = self.input(name="Reader")
         return self.base_define_graph(images, labels, bb)
 
-test_data = {
+test_data = {\
```
Comment (jantonguirao) [inline dali/test/python/test_detection_pipeline.py]: is that `\` intentional?

Comment (awolant) [reply dali/test/python/test_detection_pipeline.py]: No, removed it.

**Comment by awolant** [comment]
Rebuild #681097

File: dali/pipeline/operators/geometric/bb_flip.cc [hunk 2]
```diff
@@ -75,7 +75,7 @@ void BbFlip<CPUBackend>::RunImpl(dali::SampleWorkspace *ws, const int idx) {
   auto output_data = output.mutable_data<float>();
 
   for (int i = 0; i < input.size(); i += 4) {
-    auto bbox = ltrb_ ? BoundingBox::FromLtrb(&input_data[i])
+    auto bbox = ltrb_ ? BoundingBox::FromLtrb(&input_data[i], BoundingBox::NoBounds())
                       : BoundingBox::FromXywh(&input_data[i]);
```
Comment (mzient) [inline dali/pipeline/operators/geometric/bb_flip.cc]: Why the double standard here? It's more likely that an XYWH box will be out of bounds than LTRB due to rounding problems.

Comment (awolant) [reply dali/pipeline/operators/geometric/bb_flip.cc]: Done

**Comment by awolant** [comment]
Rebuild #682159

**Pull request merged**